Odata-docs/client/v6/basic-crud-operations.md

```diff
@@ -74,3 +74,29 @@ context.DeleteObject(person); // create a delete request
 
 context.SaveChanges(); // send the request
 ```
+
+## Handle Client Exceptions
+OData Client has implemented a number of `Exception` classes including `DataServiceQueryException` and `DataServiceClientException`. Below is an example on how Exception can be handled while using OData client.
+
+``` csharp
+ try
+ {
+    //Code that throws exception from the service
+ }
+ catch (DataServiceQueryException ex)
+ {
+     //Client level Exception message
+     Console.WriteLine(ex.Message);
+
+     //The InnerException of DataServiceQueryException contains DataServiceClientException
+     DataServiceClientException dataServiceClientException = ex.InnerException as DataServiceClientException;
```

Comment:
Is inner exception always set or are there instances where it could be null?

Reply:
I get your point. It could be `null`.